### PR TITLE
fix(cli/repl): Fixing syntax highlighting

### PR DESCRIPTION
**Pull request by AliBasicCoder**
this pull request fixes syntax highlighting by doing so:
1. doesn't highlight numbers between letters like `Uint8Array`
1. highlight binary numbes like `0b10001` and don't highlight invalid ones
1. highlight octal numbes like `0o707` and don't highlight invalid ones
1. highlight hexadecimal numbes like `0xfff` and don't highlight invalid ones
1. highlight keywords `this`, `NaN` & `Infinity`

before:
before

after:
after

### CODE REVIEW

**Comment by bartlomieju** [comment]
@caspervonb please review when you get a chance

**Comment by pedropaulosuzuki** [comment]
Template string values are also interpreted as usual strings:

image

File: cli/repl.rs [hunk 1]
```diff
@@ -222,11 +222,15 @@ impl LineHighlighter {
...
+      (?P<number>\b\d+(?:\.\d+)?(?:e[+-]?\d+)*n?\b) |
+      (?P<infinity>\b(?:Infinity|NaN)\b) |
+      (?P<hexnumber>\b0x[a-fA-F0-9]+\b) |
+      (?P<octalnumber>\b0o[0-7]+\b) |
+      (?P<binarynumber>\b0b[01]+\b) |
       (?P<boolean>\b(?:true|false)\b) |
       (?P<null>\b(?:null)\b) |
       (?P<undefined>\b(?:undefined)\b) |
-      (?P<keyword>\b(?:await|async|var|let|for|if|else|in|of|class|const|function|yield|return|with|case|break|switch|import|export|new|while|do|throw|catch)\b) |
+      (?P<keyword>\b(?:await|async|var|let|for|if|else|in|of|class|const|function|yield|return|with|case|break|switch|import|export|new|while|do|throw|catch|this)\b) |
```
Comment (caspervonb) [inline cli/repl.rs:233]: This was left out on purpose.

Editors tend to disagree here so up for debate, but I personally at-least tend to not highlight intrinsic values and identifiers.

Comment (AliBasicCoder) [reply cli/repl.rs:233]: > Editors tend to disagree here

well, nano and vim treat this as a keyword (meaning giving it the same color as keywords)
sublime text and vscode gives it a special color
and all text editors (as i know) where cli or not highlight "this" in some way

nano:
nano
sublime:
sumblime_text
vim:
vim
vscode:
vscode

>  but I personally at-least tend to not highlight intrinsic values and identifiers. If we do it too much the entire output lights up a bright yellow

highlighting "this" only is not too much

Comment (bartlomieju) [reply cli/repl.rs:233]: I agree with @AliBasicCoder on this one - I've never seen `this` not being highlighted.

**Comment by bartlomieju** [comment]
> Template string values are also interpreted as usual strings:
> 
> image

@AliBasicCoder does your PR fix this problem as well?

**Comment by AliBasicCoder** [comment]
> > Template string values are also interpreted as usual strings:
> > 
> > image
> 
> @AliBasicCoder does your PR fix this problem as well?

no

**Review by crypt096 — APPROVED** [review]

**Comment by AliBasicCoder** [comment]
@caspervonb so? this PR is approved??

**Comment by caspervonb** [comment]
> @caspervonb so? this PR is approved??

@bartlomieju has to sign off on it but I have no objections, LGTM.

I'd like to see "_this_" in a different color group but I plan to be combing through the colour palette used across different subcommands soon'ish.

**Review by kitsonk — APPROVED** [review]
LGTM

Disappointed we don't have some sorts of tests for this. 😢

**Pull request merged**